cgen error for `dump(x?.str())` when x is `?&html.Tag` (with the default `-skip-unused`; it works fine with `-no-skip-unused`)

<!-- It is advisable to update all relevant modules using `v outdated` and `v install` -->

<details>
<summary>V version: V 0.4.9 ee66c72, press to see full `v doctor` output</summary>

|V full version      |V 0.4.9 bd10e12.ee66c72
|:-------------------|:-------------------
|OS                  |linux, Ubuntu 20.04.6 LTS
|Processor           |4 cpus, 64bit, little endian, Intel(R) Core(TM) i3-3225 CPU @ 3.30GHz
|Memory              |3.55GB/15.05GB
|                    |
|V executable        |/home/delian/code/v/v
|V last modified time|2025-01-23 10:23:44
|                    |
|V home dir          |OK, value: /home/delian/code/v
|VMODULES            |OK, value: /home/delian/.vmodules
|VTMP                |OK, value: /tmp/v_1000
|Current working dir |OK, value: /home/delian/code/v
|                    |
|Git version         |git version 2.47.1
|V git status        |weekly.2025.03-26-gee66c72a (1 commit(s) behind V master)
|.git/config present |true
|                    |
|cc version          |cc (Ubuntu 10.5.0-1ubuntu1~20.04) 10.5.0
|gcc version         |gcc (Ubuntu 10.5.0-1ubuntu1~20.04) 10.5.0
|clang version       |clang version 10.0.0-4ubuntu1 
|tcc version         |tcc version 0.9.28rc 2024-07-31 HEAD@1cee0908 (x86_64 Linux)
|tcc git status      |thirdparty-linux-amd64 0134e9b9
|emcc version        |N/A
|glibc version       |ldd (Ubuntu GLIBC 2.31-0ubuntu9.16) 2.31

</details>

**What did you do?**
`./v -g -o vdbg cmd/v && ./vdbg  bug3.v && bug3`
```v
module main

import net.html

fn main() {
	mut doc := html.parse('<body><div class="Box-footer"><div class="Truncate">abc</div></div></body>')
	footer := doc.get_tags_by_class_name('Box-footer')[0]
	hrefs := footer.get_tag_by_class_name('Truncate')
	dump(hrefs?.str()) // cgen error
}


```

**What did you see?**
```
================== C compilation error (from tcc): ==============
cc: /tmp/v_1000/bug3.01JJ9D69H38WBHB1TDGJ9PYNAT.tmp.c:8701: warning: implicit declaration of function 'isnil'
cc: /home/delian/code/v/thirdparty/tcc/lib/libgc.a: error: 'GC_noop1_ptr' defined twice
cc: tcc: error: undefined symbol 'isnil'
=================================================================
(You can pass `-cg`, or `-show-c-output` as well, to print all the C error messages).
```

**What did you expect to see?**

a compiled program, even with -skip-unused (the default)


> [!NOTE]
> You can use the 👍 reaction to increase the issue's priority for developers.
>
> Please note that only the 👍 reaction to the issue itself counts as a vote.
> Other reactions and those to comments will not be taken into account.

--- a/vlib/v/checker/fn.v
+++ b/vlib/v/checker/fn.v
@@ -759,7 +759,7 @@ fn (mut c Checker) call_expr(mut node ast.CallExpr) ast.Type {
 		c.markused_option_or_result(!c.is_builtin_mod && c.mod != 'strings')
 	}
 	c.expected_or_type = old_expected_or_type
-	c.markused_call_expr(mut node)
+	c.markused_call_expr(left_type, mut node)
 	if !c.inside_const && c.table.cur_fn != unsafe { nil } && !c.table.cur_fn.is_main
 		&& !c.table.cur_fn.is_test {
 		// TODO: use just `if node.or_block.kind == .propagate_result && !c.table.cur_fn.return_type.has_flag(.result) {` after the deprecation for ?!Type

--- a/vlib/v/checker/used_features.v
+++ b/vlib/v/checker/used_features.v
@@ -111,7 +111,7 @@ fn (mut c Checker) markused_comptimefor(mut node ast.ComptimeFor, unwrapped_expr
 	}
 }
 
-fn (mut c Checker) markused_call_expr(mut node ast.CallExpr) {
+fn (mut c Checker) markused_call_expr(left_type ast.Type, mut node ast.CallExpr) {
 	if !c.is_builtin_mod && c.mod == 'main' && !c.table.used_features.external_types {
 		if node.is_method {
 			if c.table.sym(node.left_type).is_builtin() {
@@ -120,6 +120,10 @@ fn (mut c Checker) markused_call_expr(mut node ast.CallExpr) {
 		} else if node.name.contains('.') {
 			c.table.used_features.external_types = true
 		}
+	}
+	if left_type != 0 && left_type.is_ptr() && !c.table.used_features.auto_str_ptr
+		&& node.name == 'str' {
+		c.table.used_features.auto_str_ptr = true
 	}
 }
 

--- a/vlib/v/tests/skip_unused/option_ptr_print.run.out
+++ b/vlib/v/tests/skip_unused/option_ptr_print.run.out
@@ -1,0 +1,1 @@
+[vlib/v/tests/skip_unused/option_ptr_print.vv:7] hrefs.str(): &<div class="Truncate" >abc</div>

--- a/vlib/v/tests/skip_unused/option_ptr_print.skip_unused.run.out
+++ b/vlib/v/tests/skip_unused/option_ptr_print.skip_unused.run.out
@@ -1,0 +1,1 @@
+[vlib/v/tests/skip_unused/option_ptr_print.vv:7] hrefs.str(): &<div class="Truncate" >abc</div>

--- a/vlib/v/tests/skip_unused/option_ptr_print.vv
+++ b/vlib/v/tests/skip_unused/option_ptr_print.vv
@@ -1,0 +1,8 @@
+import net.html
+
+fn main() {
+	mut doc := html.parse('<body><div class="Box-footer"><div class="Truncate">abc</div></div></body>')
+	footer := doc.get_tags_by_class_name('Box-footer')[0]
+	hrefs := footer.get_tag_by_class_name('Truncate')
+	dump(hrefs?.str())
+}

